gabarit/template_nlp/nlp_project/package_name/models_training/model_huggingface.py

```diff
@@ -0,0 +1,741 @@
+        '''Tokenizes input data
+
+        Args:
+            examples (Batch): input data (Dataset Batch)
+        Returns:
+            BatchEncoding: tokenized data
+        '''
+        # Padding to False as we will use a Trainer and a DataCollatorWithPadding that will manage padding for us (better limit the memory impact)
+        # We leave max_length to None -> backup on model max length
+        # https://stackoverflow.com/questions/74657367/how-do-i-know-which-parameters-to-use-with-a-pretrained-tokenizer
+        return self.tokenizer(examples["text"], padding=False, truncation=True)
+
+    def _get_model(self, model_path: str = None, num_labels: int = None) -> Any:
+        '''Gets a model structure - returns the instance model instead if already defined
+
+        Returns:
+            (Any): a HF model
+        '''
+        # Return model if already set
+        if self.model is not None:
+            return self.model
+
+        model = AutoModelForSequenceClassification.from_pretrained(
+                self.transformer_name if model_path is None else model_path,
+                num_labels=len(self.list_classes) if num_labels is None else num_labels,
+                problem_type="multi_label_classification" if self.multi_label else "single_label_classification",
+                cache_dir=HF_CACHE_DIR)
+        # Set model on gpu if available
+        model = model.to('cuda') if self._is_gpu_activated() else model.to('cpu')
+        return model
+
+    def _get_tokenizer(self, model_path: str = None) -> PreTrainedTokenizer:
+        '''Gets a tokenizer
+
+        Returns:
+            (PreTrainedTokenizer): a HF tokenizer
+        '''
+        # Return tokenizer if already set
+        if self.tokenizer is not None:
+            return self.tokenizer
+
+        tokenizer = AutoTokenizer.from_pretrained(self.transformer_name if model_path is None else model_path, cache_dir=HF_CACHE_DIR)
+        return tokenizer
+
+    def _get_optimizers(self) -> Tuple[Any, Any]:
+        '''Fonction to define the Trainer optimizers
+           -> per default return (None, None), i.e. default optimizers (cf HF Trainer doc)
+
+        Returns:
+            Tuple (Optimizer, LambdaLR): An optimizer/scheduler couple
+        '''
+        # e.g.
+        # Here, your custom Optimizer / scheduler couple
+        # (check https://huggingface.co/docs/transformers/v4.24.0/en/main_classes/optimizer_schedules)
+        return (None, None)
+
+    def _compute_metrics_mono_label(self, eval_pred: EvalPrediction) -> dict:
+        '''Computes some metrics for mono label cases
+
+        Args:
+            eval_pred: predicted & ground truth values to be considered
+        Returns:
+            dict: dictionnary with computed metrics
+        '''
+        # Load metrics
+        metric_accuracy = load_metric(os.path.join(os.path.dirname(os.path.abspath(__file__)), "hf_metrics", "accuracy.py"))
```

Comment:
It might cleaner to do :

```python
from .hf_metrics import accuracy, precision, recall, f1

metric_accuracy = load_metric(accuracy.__file__)
```